### PR TITLE
Rename nodereport module to node-report

### PR DESCRIPTION
**Pull request by rnchamberlain**
Provides code, documentation, test and demo changes required for the migration of the nodereport project to node-report. The main changes are in the build, tests and demos that use the module name, and in the name of the report file. Remaining changes are mostly in comments. Internal variable and macro names are not changed.

This supercedes https://github.com/nodejs/node-report/pull/43

CI: https://ci.nodejs.org/view/post-mortem/job/nodereport-continuous-integration/44/

### CODE REVIEW

**Comment by rnchamberlain** [comment]
@mhdawson @richardlau this PR is almost identical to https://github.com/nodejs/node-report/pull/43, would you mind reviewing again - many thanks!

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "nodereport",
+  "name": "node-report",
   "version": "1.0.7",
-  "description": "Diagnostic NodeReport",
-  "homepage": "https://github.com/nodejs/nodereport#readme",
+  "description": "Diagnostic Report for Node",
```
Comment (richardlau) [inline package.json]: s/Node/Node.js

File: LICENCE.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 The MIT License (MIT)
 =====================
 
-Copyright (c) 2016 nodereport contributors
+Copyright (c) 2016 node-report contributors
 --------------------------------------------------
 
 *nodereport contributors listed at <https://github.com/nodejs/nodereport/blob/master/AUTHORS>*
```
Comment (rnchamberlain) [inline LICENCE.md]: fixed

File: README.md [hunk 3]
```diff
@@ -66,14 +66,14 @@ nodereport.triggerReport("myReportName");
...
-nodereport.setEvents("exception+fatalerror+signal+apicall");
-nodereport.setSignal("SIGUSR2|SIGQUIT");
-nodereport.setFileName("stdout|stderr|<filename>");
-nodereport.setDirectory("<full path>");
-nodereport.setVerbose("yes|no");
+setEvents("exception+fatalerror+signal+apicall");
+setSignal("SIGUSR2|SIGQUIT");
+setFileName("stdout|stderr|<filename>");
+setDirectory("<full path>");
+setVerbose("yes|no");
```
Comment (rnchamberlain) [inline README.md]: good idea, done

File: demo/api_call.js [hunk 4]
```diff
@@ -8,12 +8,12 @@ function my_listener(request, response) {
   switch(count++) {
   case 0:
     response.writeHead(200,{"Content-Type": "text/plain"});
-    response.write("\nRunning NodeReport API demo... refresh page to trigger NodeReport");
+    response.write("\nRunning node-report API demo... refresh page to trigger NodeReport");
```
Comment (rnchamberlain) [inline demo/api_call.js]: fixed

File: demo/loop.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ function my_listener(request, response) {
   switch(count++) {
   case 0:
     response.writeHead(200,{"Content-Type": "text/plain"});
-    response.write("\nRunning NodeReport looping application demo. Node process ID = " + process.pid);
+    response.write("\nRunning node-report looping application demo. Node process ID = " + process.pid);
     response.write("\n\nRefresh page to enter loop, then use 'kill -USR2 " + process.pid + "' to trigger NodeReport");
```
Comment (rnchamberlain) [inline demo/loop.js]: fixed

File: package.json (hunk 1 above)
Comment (rnchamberlain) [inline package.json]: fixed

**Comment by rnchamberlain** [comment]
Landed as https://github.com/nodejs/node-report/commit/a590cd9cd0445f9cb66b39979f654a0cc71eb0d1

**Pull request closed without merging**